fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -19,6 +23,9 @@ use super::{
     FvmMessage, FvmMessageInterpreter,
 };
 
+/// Indicates whether the node has been frozen due to not having a required upgrade.
+pub static IS_FROZEN: AtomicBool = AtomicBool::new(false);
```

Comment:
Can this live inside the `FvmMessageInterpreter` instead of being a public static global field? The way it is, it can be set from anywhere, by anyone. There has been no rule so far that there can be only one interpreter instance. The application only creates one, so there is no problem with multiple conflicting values.